Feature/content provider volatile

qiubix opened the pull request:
This ContentProvider doesn't provide persistance yet.

qiubix committed:
Add first test for Content provider

qiubix committed:
Fix test for empty list

qiubix committed:
Add test for adding item

qiubix committed:
Create separate test suite for MovieList

qiubix committed:
Add isEmpty method

qiubix committed:
Test for adding movie entry

qiubix committed:
introduce hasMovie helper variable

qiubix committed:
Introduce size method

qiubix committed:
common contentProvider for tests

shouldReturnSizeTwoForContentProviderWithTwoMovies

qiubix committed:
Get entry by id

qiubix committed:
store movies in hashmap

qiubix committed:
Get entry by title

qiubix committed:
Introduce singleton

clean up MovieListTest

qiubix committed:
Fix MovieListTest

tempaowca commented on app/src/main/java/sleepyweasel/purplefluffernutter/MovieEntry.java:
```diff
@@ -0,0 +1,18 @@
+
+    private String title;
+
+    public MovieEntry(String title) {
```
@AllArgsConstructor (z lomboka, na klase, a to wywalic)

qiubix replied:
jak to się robi? :(

tempaowca replied:
tak jak napisałem ;)

tempaowca commented on app/src/main/java/sleepyweasel/purplefluffernutter/MovieEntry.java:
```diff
@@ -0,0 +1,18 @@
+        this.title = title;
+    }
+
+    public String getTitle() {
```
@Getter (z lomboka, na klase, a to wywalic)

tempaowca commented on app/src/main/java/sleepyweasel/purplefluffernutter/MovieEntry.java:
```diff
@@ -0,0 +1,18 @@
+    public String getTitle() {
+        return title;
+    }
+
```
@ToString (z lomboka, na klase, a to wywalic)

tempaowca commented on app/src/test/java/sleepyweasel/purplefluffernutter/ContentProviderTest.java:
```diff
@@ -0,0 +1,105 @@
+
+    @Test
+    public void shouldReturnSizeZeroIfContentProviderEmpty() throws Exception {
+        assertThat(contentProvider.size()).isEqualTo(0);
```
isZero()

tempaowca replied:
i tu nie wiesz w sumie że jest pusty, bo to sprawdza test inny.

qiubix replied:
to jak mam to zmienić, poza tym, że isZero()? Przed każdym testem tworzony jest na początku pusty provider

tempaowca commented on app/src/test/java/sleepyweasel/purplefluffernutter/ContentProviderTest.java:
```diff
@@ -0,0 +1,105 @@
+
+    @Test
+    public void shouldCreateEmptyContentProvider() throws Exception {
+        assertThat(contentProvider.isEmpty()).isTrue();
```
ten test sprawdza trochę co innego.

tempaowca commented on app/src/main/java/sleepyweasel/purplefluffernutter/MovieContentProvider.java:
```diff
@@ -0,0 +1,64 @@
+
+    private int nextId;
+
+    private HashMap<Integer,MovieEntry> movieList;
```
to nie jest lista w żadnym wypadku. może po prostu movies?

tempaowca commented on app/src/test/java/sleepyweasel/purplefluffernutter/ContentProviderTest.java:
```diff
@@ -0,0 +1,105 @@
+
+        MovieEntry entry = contentProvider.getEntryByTitle(MOVIE_TITLE_1);
+
+        assertThat(entry.getTitle()).isEqualTo(MOVIE_TITLE_1);
```
a tu ni elepiej obiekty?

tempaowca commented on app/src/test/java/sleepyweasel/purplefluffernutter/ContentProviderTest.java:
```diff
@@ -0,0 +1,105 @@
+
+        MovieEntry entry = contentProvider.getEntry(1);
+
+        assertThat(entry.getTitle()).isEqualTo(MOVIE_TITLE_2);
```
a tu nie lepiej obiekty?

tempaowca commented on app/src/test/java/sleepyweasel/purplefluffernutter/ContentProviderTest.java:
```diff
@@ -0,0 +1,105 @@
+
+        MovieEntry entry = contentProvider.getEntry(0);
+
+        assertThat(entry.getTitle()).isEqualTo(MOVIE_TITLE_1);
```
a tu nie lepiej obiekty?

qiubix committed:
Refactor after review

qiubix committed:
Lombok constructor for MovieEntry

qiubix committed:
Add movie to content provider on save

fix application flow: back button from MovieDetail leads to MovieList,
not one level up.

qiubix committed:
Fix movie label on list

tempaowca commented on app/src/main/java/sleepyweasel/purplefluffernutter/EditMovieActivity.java:
```diff
@@ -51,6 +51,18 @@ public void goToList(View v) {
         startActivity(i);
     }
 
+    public void onClickSaveButton(View v) {
```
mamy juz butter knife, to moze podpinajmy @OnClickiem

qiubix replied:
nie umiem tego podpiąć. Usunąłem ten onClick z xmla oraz dodałem @OnClick(R.id.button_save), ale teraz naciśnięcie przycisku powoduje, że nic się nie dzieje. (View v) powinno zostać jako argument, prawda?

qiubix replied:
dobra, jestem głupi. już umiem

tempaowca replied:
zobacz jak ja to podpialem. tam w oncreate musisz taka metode jeszcze odpalic.

tempaowca commented on app/src/main/java/sleepyweasel/purplefluffernutter/EditMovieActivity.java:
```diff
@@ -51,6 +51,18 @@ public void goToList(View v) {
+    }
+
+    public void addNewMovieToContentProvider() {
+        TextView text = (TextView) findViewById(R.id.movie_title_value);
```
to też butterem możesz

tempaowca commented on app/src/main/java/sleepyweasel/purplefluffernutter/EditMovieActivity.java:
```diff
@@ -51,6 +51,18 @@ public void goToList(View v) {
+        TextView text = (TextView) findViewById(R.id.movie_title_value);
+        String movieTitle = text.getText().toString();
+        MovieEntry movieEntry = new MovieEntry(movieTitle);
+        MovieContentProvider.getInstance().addMovie(movieEntry);
```
no tu mi sie nadal nie podoba, że on jest ciagniety instancja zamiast byc wstrzyknietym do tego activity.

qiubix replied:
to jak mam go wstrzyknąć? Konstruktorem?

tempaowca replied:
albo podepnij daggera 2 albo olej. Lepiej olej. ja to najwyzej zrobie jak sie skusze na podpiecie.

tempaowca commented on app/src/main/java/sleepyweasel/purplefluffernutter/MovieContentProvider.java:
```diff
@@ -0,0 +1,64 @@
+        nextId = 0;
+    }
+
+    public ArrayList<String> getArrayList() {
```
ta nazwa mnie przeraża.

qiubix replied:
hm.. rzeczywiście jest beznadziejna. Generalnie chodzi o to, że w tym ListAdapterze, który tworzę w MovieListFragment trzeba mu podać jako argument listę, żeby sobie przy jej pomocy spopulował widok. Jakiś pomysł na lepszą nazwę?

tempaowca replied:
getMovieTitles np.

qiubix replied:
okazuje się, że sam jestem w stanie wymyślić nazwę, bo wpadłem na taką samą, jak pomyślałem chwilę. Jednak programowanie w nocy zostawia pewne żniwo. Ale za to uwagi głównie do refaktoringu ;)

tempaowca commented on app/src/main/java/sleepyweasel/purplefluffernutter/MovieEntry.java:
```diff
@@ -0,0 +1,16 @@
+
+    private String title;
+
+    public String toString() {
```
a @ToString czemu nie?

qiubix replied:
było @ToString, ale cofnąłem. Dodaj sobie i odpal testy, a zobaczysz ;)

tempaowca commented on app/src/test/java/sleepyweasel/purplefluffernutter/EditMovieActivityTest.java:
```diff
@@ -0,0 +1,29 @@
+
+        saveButton.performClick();
+
+        assertThat(contentProvider.isEmpty()).isFalse();
```
może tu lepiej sprawdzić, że dodał się jeden i to dokładnie ten co chcemy. ogólnie te size() i isEmpty() sa tylko w testach uzywane, ale moze cos przeoczyłem.

qiubix replied:
chyba rzeczywiście tylko w testach, jak na razie, ale size() może być używany później do wielu rzeczy, zresztą wydaje mi się, że akurat w przypadku contentProvidera jego interfejs powinien je zawierać.

qiubix replied:
najwyżej później je zaorasz ;)

tempaowca replied:
tak czy owak sprawdz ze sie dodalo to co trzeba.

tempaowca commented on app/src/test/java/sleepyweasel/purplefluffernutter/MovieListTest.java:
```diff
@@ -0,0 +1,58 @@
+
+    private MovieContentProvider contentProvider;
+
+    private final String MOVIE_TITLE = "Movie title";
```
static może być

tempaowca commented:
No i ogólnie jeszcze, skoro już można coś dodawać to te dummy są do czegoś potrzebne jeszcze?

qiubix commented:
te dummy już nie są potrzebne, ale jeszcze ich nie kasowałem, bo tam chciałem podpatrzeć jakieś przekazywanie tytułu między fragmentami itd, nie wiem jeszcze jak to robić dobrze.

tempaowca commented on app/src/main/java/sleepyweasel/purplefluffernutter/MovieContentProvider.java:
```diff
@@ -0,0 +1,64 @@
+    public ArrayList<String> getArrayList() {
+        ArrayList<String> arrayList = new ArrayList<>();
+        for (MovieEntry movie : movies.values()) {
+            arrayList.add(movie.toString());
```
a tu nie powinno być w sumie getTitle?

tempaowca commented on app/src/test/java/sleepyweasel/purplefluffernutter/MovieListTest.java:
```diff
@@ -0,0 +1,58 @@
+        ListAdapter adapter = getListAdapter();
+        String label = adapter.getItem(0).toString();
+
+        assertThat(label).isEqualTo(MOVIE_TITLE);
```
no a dlaczego tutaj wlasciwie item.toString ma byc rowny tytulowi?

qiubix committed:
Use ButterKnife, fix method names, use getTitle for MovieEntry

qiubix committed:
Enhance test for adding movie on save

cleanup, use butterknife, check whether values are correct
